math/src/field/smallprimefield/generic_prime_field_elt.rs

```diff
@@ -0,0 +1,426 @@
+use crate::field::{
+    traits::{FieldElement, StarkField},
+    QuadExtensionA,
+};
+use core::{
+    convert::{TryFrom, TryInto},
+    fmt::{Debug, Display, Formatter},
+    mem,
+    ops::{Add, AddAssign, Div, DivAssign, Mul, MulAssign, Neg, Sub, SubAssign},
+    slice,
+};
+use utils::{
+    collections::Vec,
+    string::{String, ToString},
+    AsBytes, ByteReader, ByteWriter, Deserializable, DeserializationError, Randomizable,
+    Serializable,
+};
+
+// Number of bytes needed to represent field element
+const ELEMENT_BYTES: usize = mem::size_of::<u64>();
+
+#[derive(Copy, Clone, Debug, PartialEq, Eq, Default)]
```

Comment:
Would be great to add some comments here (nothing fancy) about the purpose of this field etc.